.github/workflows/frontend.yml

```
@@ -0,0 +1,40 @@
+      - name: Run frontend tests
+        - name: Run frontend tests
+          run: |
+            if [ -f frontend/package.json ]; then
+              cd frontend
+              if node -e 'const pkg=require("./package.json"); process.exit(pkg.scripts && pkg.scripts.test ? 0 : 1)' ; then
+                pnpm test
+              else
+                echo "No frontend tests defined; skipping"
+              fi
+            fi
+
```

YAMLlint (1.37.1)

[warning] too many blank lines (1 > 0)

(empty-lines)